packages/zsa/src/zod-safe-function.ts

```diff
@@ -638,8 +638,19 @@ export class ZodSafeFunction<
     }
 
     // finally return the error
+    let inputSchema = await this.evaluateInputSchema({
+      ctx: undefined as any,
+      opts: inputRaw,
+      noFunctionsAllowed: false,
+    })
```

Comment:
Is this ok? All tests pass and my code works fine. Do you see any issues? `noFunctionsAllowed` is a bit scary : )

Reply:
Concur. Just for background, the reason it is there is because `zsa-openapi` offers a way to [generate open api documents](https://zsa.vercel.app/docs/configuring-openapi#openapi-documentation) from the input/output schemas. Since the input schemas can now be functions, we actually can't generate documents for users who use functions because that would require actually executing the actions and procedures which is not appropriate in this case. Hopefully that makes sense, I agree it is a bit janky but worth the addition of input schema functions.

Reply:
Agree, is a bit weird but worth the addition. 

Maybe doc generation could receive a fake context. This way inputs could be functions.